mixnode/src/commands/sign.rs

```diff
@@ -3,39 +3,44 @@
 
 use crate::commands::*;
 use crate::config::{persistence::pathfinder::MixNodePathfinder, Config};
-use clap::{App, Arg, ArgMatches};
-use colored::Colorize;
+use clap::ArgGroup;
 use config::NymConfig;
 use crypto::asymmetric::identity;
 use log::error;
 
-const SIGN_TEXT_ARG_NAME: &str = "text";
-const SIGN_ADDRESS_ARG_NAME: &str = "address";
-
-pub fn command_args<'a, 'b>() -> App<'a, 'b> {
-    App::new("sign")
-        .about("Sign text to prove ownership of this mixnode")
-        .arg(
-            Arg::with_name(ID_ARG_NAME)
-                .long(ID_ARG_NAME)
-                .help("The id of the mixnode you want to sign with")
-                .takes_value(true)
-                .required(true),
-        )
-        .arg(
-            Arg::with_name(SIGN_ADDRESS_ARG_NAME)
-                .long(SIGN_ADDRESS_ARG_NAME)
-                .help("Signs your blockchain address with your identity key")
-                .takes_value(true)
-                .conflicts_with(SIGN_TEXT_ARG_NAME),
-        )
-        .arg(
-            Arg::with_name(SIGN_TEXT_ARG_NAME)
-                .long(SIGN_TEXT_ARG_NAME)
-                .help("Signs an arbitrary piece of text with your identity key")
-                .takes_value(true)
-                .conflicts_with(SIGN_ADDRESS_ARG_NAME),
-        )
+#[derive(Args, Clone)]
+#[clap(group(ArgGroup::new("sign").required(true).args(&["address", "text"])))]
+pub(crate) struct Sign {
+    /// The id of the mixnode you want to sign with
+    #[clap(long)]
+    id: String,
+
+    /// Signs your blockchain address with your identity key
+    #[clap(long)]
+    address: Option<String>,
+
+    /// Signs an arbitrary piece of text with your identity key
+    #[clap(long)]
+    text: Option<String>,
+}
+
+enum SignedTarget {
+    Text(String),
+    Address(String),
+}
+
+impl From<Sign> for SignedTarget {
+    fn from(args: Sign) -> Self {
+        if let Some(text) = args.text {
+            SignedTarget::Text(text)
+        } else if let Some(address) = args.address {
+            SignedTarget::Address(address)
+        } else {
+            // Clap should guarantee this by using ArgGroup. Perhaps support for outputting the enum
```

Comment:
I only had a very brief look at the clap's macro stuff, so my suggestion might be completely wrong. But perhaps `ArgEnum` might solve this issue here?